bedwars-plugin/src/main/java/com/tomkeuper/bedwars/listeners/DamageDeathMove.java

```diff
@@ -85,60 +89,71 @@ public DamageDeathMove() {
-                    e.setCancelled(true);
-                    return;
-                }
-                if (a.isReSpawning(p)) {
-                    e.setCancelled(true);
-                    return;
-                }
-                if (a.getStatus() != GameState.playing) {
-                    e.setCancelled(true);
-                    return;
-                }
+        World world = e.getEntity().getLocation().getWorld();
 
-                if (BedWarsTeam.reSpawnInvulnerability.containsKey(p.getUniqueId())) {
-                    if (BedWarsTeam.reSpawnInvulnerability.get(p.getUniqueId()) > System.currentTimeMillis()) {
-                        e.setCancelled(true);
-                    } else BedWarsTeam.reSpawnInvulnerability.remove(p.getUniqueId());
-                }
-            }
-        }
         if (BedWars.getServerType() == ServerType.MULTIARENA) {
-            if (e.getEntity().getLocation().getWorld().getName().equalsIgnoreCase(BedWars.getLobbyWorld())) {
+            if (world.getName().equalsIgnoreCase(BedWars.getLobbyWorld())) {
                 e.setCancelled(true);
+                return;
             }
         }
+
+        if (!(e.getEntity() instanceof Player)) return;
+
+        Player player = (Player) e.getEntity();
+        IArena arena = Arena.getArenaByPlayer(player);
+
+        if (arena == null) return;
+
+        if (arena.isSpectator(player)) {
+            e.setCancelled(true);
+            return;
+        }
+
+        if (arena.isReSpawning(player)) {
+            e.setCancelled(true);
+            return;
+        }
+
+        if (arena.getStatus() != GameState.playing) {
+            e.setCancelled(true);
+            return;
+        }
+
+        // protection after re-spawn
+        if (BedWarsTeam.reSpawnInvulnerability.containsKey(player.getUniqueId())) {
+            if (BedWarsTeam.reSpawnInvulnerability.get(player.getUniqueId()) > System.currentTimeMillis()) e.setCancelled(true);
+            else BedWarsTeam.reSpawnInvulnerability.remove(player.getUniqueId());
+        }
     }
 
     //Todo show player health on bow hit
     @EventHandler(priority = EventPriority.MONITOR)
     public void onBowHit(EntityDamageByEntityEvent e) {
-        if(e.isCancelled()) return;
+        if (e.isCancelled()) return;
         if (e.getEntity().getType() != EntityType.PLAYER) return;
         if (!(e.getDamager() instanceof Projectile)) return;
+
         Projectile projectile = (Projectile) e.getDamager();
         if (projectile.getShooter() == null) return;
         if (!(projectile.getShooter() instanceof Player)) return;
 
-        Player p = (Player) e.getEntity();
+        Player player = (Player) e.getEntity();
         Player damager = (Player) projectile.getShooter();
-        IArena a = Arena.getArenaByPlayer(p);
-        if (a == null) return;
-        if (a.getStatus() != GameState.playing) return;
+        IArena arena = Arena.getArenaByPlayer(player);
```

Comment:
If we change a to arena here then we should change it for all variable names. In my opinion using full variable names is better so this would be the way to go. But in other classes you are using the shorter version again.